args.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package squalor
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var baseTypes = map[reflect.Type]bool{
+	reflect.TypeOf(bool(true)): true,
+	reflect.TypeOf(int(0)):     true,
+	reflect.TypeOf(int8(0)):    true,
+	reflect.TypeOf(int16(0)):   true,
+	reflect.TypeOf(int32(0)):   true,
+	reflect.TypeOf(int64(0)):   true,
+	reflect.TypeOf(uint(0)):    true,
+	reflect.TypeOf(uint8(0)):   true,
+	reflect.TypeOf(uint16(0)):  true,
+	reflect.TypeOf(uint32(0)):  true,
+	reflect.TypeOf(uint64(0)):  true,
+	reflect.TypeOf(float32(0)): true,
+	reflect.TypeOf(float64(0)): true,
+	reflect.TypeOf(string("")): true,
```

Comment:
Change this from `reflect.TypeOf(new(string))` to get the type `string`, rather than `*string`. In particular when instantiating a locus in `initScan`, we need to refer to the type itself.